docs/theming.md

```diff
@@ -109,15 +109,91 @@ dark theme.
 In order to style your own components with our tooling, the component's styles must be defined 
 with Sass. 
 
-You can consume the theming functions and variables from the `@angular/material/core/theming`.
-You can use the `md-color` function to extract a specific color from a palette. For example:
+You can consume the theming functions and variables from the @angular/material/core/theming. You can use the `map-get` function to extract the theming variables and `md-color` function to extract a specific color from a palette. For example:
+
+app/custom-input/custom-input-theme.scss
+
 ```scss
-.unicorn-carousel {
-  background-color: md-color($primary);
-  color: md-color($primary, default-contrast);
+@import '~@angular/material/core/theming/theming';
+
+@mixin custom-input-theme($theme) {
```

Comment:
It's not really necessary to include this much css. We can communicate the idea with much less code, e.g.:
```scss
// Define a mixin that accepts a theme and outputs the color styles for the component.
@mixin candy-carousel-theme($theme) {
  // Extract whichever individual palettes you need from the theme.
  $primary: map-get($theme, primary);
  $accent: map-get($theme, accent);

  // Use md-color to extract individual colors from a palette as necessary.
  .candy-carousel {
    background-color: md-color($primary);
    border-color: md-color($accent, A400);
  }
}
```